great_expectations/data_context/store/tuple_store_backend.py

```diff
@@ -557,19 +557,35 @@ def _get_all(self) -> list[Any]:
         s3_object_keys = [self._build_s3_object_key(key) for key in keys]
         return [self._get_by_s3_object_key(client, key) for key in s3_object_keys]
 
-    def _get_by_s3_object_key(self, s3_client, s3_object_key):
+    def _get_by_s3_object_key(self, s3_client, s3_object_key) -> Any:
         try:
             s3_response_object = s3_client.get_object(Bucket=self.bucket, Key=s3_object_key)
         except (s3_client.exceptions.NoSuchKey, s3_client.exceptions.NoSuchBucket) as e:
             raise InvalidKeyError(  # noqa: TRY003 # FIXME CoP
                 f"Unable to retrieve object from TupleS3StoreBackend with the following Key: {s3_object_key!s}"  # noqa: E501 # FIXME CoP
             ) from e
 
-        return (
-            s3_response_object["Body"]
-            .read()
-            .decode(s3_response_object.get("ContentEncoding", "utf-8"))
-        )
+        # ContentEncoding is an unknown string per
+        # https://botocore.amazonaws.com/v1/documentation/api/latest/reference/services/s3/client/get_object.html#get-object
+        content_encoding: str = s3_response_object.get("ContentEncoding", "utf-8")
+        # We found the string could take the form of an array, e.g. `utf-8,aws-chunked`.
+        # As of boto3 1.36.0, we aren't aware of any time when the string-list can have more
+        # than 1 item except when `aws-chunked` is included. In order to preserve unknown
+        # encodings included with `aws-chunked` we will remove the `aws-chunked` string from
+        # the list. We do not intend to add support for reading in chunks at this time.
+        # Calling botocore.response.StreamingBody.read() without arguments
+        # will read the entire stream.
+        encodings: list[str] = content_encoding.split(",")
+        if "aws-chunked" in encodings:
+            encodings.remove("aws-chunked")
+        data = s3_response_object["Body"].read()
+        # Maybe it's theoretically possible to encode an entire file with multiple encodings.
+        # It's an odd choice of format for ContentEncoding to allow multiple encoding strings.
```

Comment:
I'm a little worried about what "as far as we know...", but this seems like it moves us forward.

Reply:
If `ContentEncoding` is empty or contains only `aws-chunked`, we will use `utf-8` like before. If there are multiple encodings other than `aws-chunked` it will likely cause an error here, but that may mean the intent of this field is something other than what the description in the API docs says. At the end of the day, this is just unfortunate.